data_steward/tools/snapshot_by_query.py

```diff
@@ -37,33 +37,64 @@ def create_empty_cdm_tables(snapshot_dataset_id):
     cdm.create_vocabulary_tables(snapshot_dataset_id)
 
 
-def get_field_cast_expr(field, data_type):
+def get_field_cast_expr(field, data_type, required_mode, existing_fields):
     """
     generates cast expression based on data_type for the field
 
     :param field: field name
     :param data_type: data type of the field
+    :param required_mode: if a field is required or nullable
+    :param existing_fields: list of field names in a table
     :return:  col string
     """
-    bigquery_int_float = {'integer': 'INT64', 'float': 'FLOAT64'}
+    bigquery_data_types = {
+        'integer': 'INT64',
+        'float': 'FLOAT64',
+        'string': 'STRING',
+        'date': 'DATE',
+        'timestamp': 'TIMESTAMP'
+    }
+    replacement_values = {
+        'integer': 0,
+        'float': 0.0,
+        'string': '\'\'',
+        'date': 'DATE(\'1970-01-01\')',
+        'timestamp': 'TIMESTAMP(\'1970-01-01\')'
+    }
+    if field not in existing_fields and required_mode == 'required':
+        col = f'CAST({replacement_values[data_type.lower()]} AS {bigquery_data_types[data_type.lower()]}) AS {field}'
+    elif field not in existing_fields and required_mode == 'nullable':
+        col = f'CAST(NULL as {bigquery_data_types[data_type.lower()]}) AS {field}'
```

Comment:
As discussed, please allow it to fail loudly if a required field is missing in the source. Something like this (untested, just for illustration):
```suggestion
    if dst_field not in existing_fields:
        if required_mode == 'required':
            raise RuntimeError(f'Unable to load the field "{dst_field}" which is required in the destination table and missing from the source table')
        else:
            col = f'CAST(NULL as {bigquery_data_types[data_type.lower()]}) AS {field}'
```